### ISSUE TITLE
Adding NHS Ethnicity codes and Admission and Discharge codes to the OMOP vocabulary 

### ISSUE THREAD

**Issue by solmazeradat** [OHDSI/Vocabulary-v5.0#843]
Hi,

As part of the Great Ormond Street Hospital (GOSH) OMOP project we would like to request the following NHS vocabularies to be added:

- NHS Ethnic Category - https://www.datadictionary.nhs.uk/data_elements/ethnic_category.html

- Admission Source - https://www.datadictionary.nhs.uk/data_elements/admission_source__hospital_provider_spell_.html?hl=admission%2Csource

- Discharge Destination - https://www.datadictionary.nhs.uk/data_elements/discharge_destination_code__hospital_provider_spell_.html?hl=discharge%2Cdestination

**Category of submission.**

- Template 4: has been used for adding the proposed new vocabularies with mappings (full or partial) to the exiting standard vocabularies.

- The templates have been added in the following location under the folder name **843_NHS_Ethnicity_Discharge_Admission_SolmazEradat** - https://drive.google.com/drive/folders/1MR_D07qaRM8w91Dk5fRUoZF2LcjBUndR?usp=drive_link 

Thanks,
Solmaz

**Comment by aostropolets** [comment]
Hi @solmazeradat! Thanks so much! Our first community contribution to go, yay! 
I looked through the submissions and I would suggest you check out the "Checklist" sheet in both submissions. They need to be filled to indicate that you went through the checks as failure to pass them would fail the submission as well. Specifically, two points for NHS Ethnic Category:

1) Failed check 13: concept_relationship_manual is missing concept_code_2 in line 10, 19 and 20. Lines 19 and 20 should simply be removed alltogether since there is no mapping. Line 10 is legitimately missing a code and we don't know what it should be.
2) Bigger one: failed check 14: some of your mappings for race map to non-standard codes from Biobank. They need to be mapped to standard codes (standard_concept = 'S') though. In fact, it points to a bigger problem here: if you want to use these source terms to populate race_concept_id you need to map them to the concepts with domain_id = 'Race' (also happen to be vocabulary_id = 'Race') as per CDM convention here: **https://ohdsi.github.io/CommonDataModel/cdm54.html#PERSON**. The convention also points to the list of accepted terms.


Could you please re-map races and upload the modified template to the same folder with _v2 in name? 
Also here if any clarifications are needed :)

**Comment by solmazeradat** [comment]
Hi @aostropolets ,

Thanks for the great feedback on the templates. 

- I have updated the checklist in both templates. 
- I also noticed I had left the synonym tab in the admission and discharge template blank, so updated this as well. 
- For the template for mapping race, I attempted to remap the concepts to standard OMOP ones. I noticed that there was no standard concept for race Caribbean. 
- I updated the _v2 on the folder. 

Happy to have a chat during this weeks upcoming call or can further discuss this during the European OHDSI Symposium next weekend which I'm very much looking forward too :) 

Thanks for all the help on this! 
Solmaz

**Comment by solmazeradat** [comment]
Hi @aostropolets,
Hope all is well.
Wanted to see if there is anything further needed on the updated documents ahead of the new release in August? 

Thanks,
Solmaz

**Comment by aostropolets** [comment]
Looking much better, thanks! We would remove 99 from Race vocab since it's this weird entity that represents "NA" and should not really be in a vocab. Otherwise looks good. One question: you mapped Mixed - White and Black Caribbean &  Mixed - White and Black African to White Race. Don't you also want to map them to Black Race?

**Comment by cgreich** [comment]
Friends:

Wrt the race and ethnicity codes: There was [long discussion before](https://forums.ohdsi.org/t/race-and-ethnicity-in-the-omop-cdm/8700). I think we ended up with saying that there shouldn't be any mapping and de-duplication. Reason: there is no rational basis for doing that. These things are social/economical/biological and cannot be deterministically defined. In other words, a Black in Brazil is not the same as a Black in the US.

Bottom line: Remove all mappings. Leave the concepts as is and make them standard. But remove all flavors of null (Not stated, Not Known). Or, if you want to keep them, de-standardize them.

**Comment by aostropolets** [comment]
Would you mind elaborating on what you propose? There is a source vocabulary with some NHS flavors of ethnicities. Are you suggest to add them to the OHDSI Vocabularies as standard terms? Also: didn't we say that community can add whatever they please as long as non-standard and doesn't break the rules?

**Comment by cgreich** [comment]
Yes. The NHS races are not the same as the US OMB races since the social context is entirely different. Therefore, they cannot be deduped. Therefore, they have to become standard. It will be up to the Analyst wretch to make conceptsets that make sense. Which I don't think will work across different societies, but that's up to them.

**Comment by aostropolets** [comment]
The process is still first add non-standard terms and then update them to standard (requires ratification of Vocabulary WG/CDM WG). So @solmazeradat can opt for option a) non-standard codes with mappings or b)non-standard w/o mappings->promotion to standard upon approval. It seems that this is for her to decide since addition of new non-standard content does not require review/approval.

**Comment by cgreich** [comment]
I see. I would strongly suggest to go through path b). If we do a) and map international race and ethnicity concepts to the American standard we will produce absurd results. Is a Black person in the UK the same thing as an African American? Is a person from Spain a Latino in the US sense? It cannot be objectively determined, since these entities have no objective definition. They are self-assigned.

> this is for her to decide

Not if the source is a public vocabulary. If it were her own private one no question. But it is adopted throughout the UK.

**Comment by aostropolets** [comment]
Setting race discussion aside, the current pathway set up by the committee is to take additions of content without a review. In this way we will take whatever Solmaz gave us (right now it's non-standard with mappings). We can set up a call to discuss it more; we need to start incorporating the files to make into the release though (and potentially bring it up to the Committee if the guidelines need to be changed).

**Comment by solmazeradat** [comment]
Hi @aostropolets and @cgreich ,

Hope you are both well. 

Apologies for not get back to you sooner on this. Just catching up after coming back annual leave. 

@aostropolets - Had a look at the upcoming changes https://github.com/OHDSI/Vocabulary-v5.0/wiki/Upcoming-changes and couldn't see the addition of the NHS codes as part of the is issue. After reading the discussions above, happy to join tomorrow's call for the first 15 minutes and chat through if any changes/support is needed in relation to adding the NHS races codes. If tomorrow's call is not suitable, can arrange a separate time. 

Many thanks,
Solmaz

**Comment by TinyRickC137** [comment]
Dear @solmazeradat, no worries, we are working on your community contribution. 

We decided to publish only the upcoming changes that may affect ETL processes, not everything that is going to be included in the release.

During processing your contributions, we encountered a couple of errors, such as:

- Different vocabulary_id in _concept_manual_ and _concept_relationship_manual_ templates: NHS Ethnic Category vs UK Ethnic Category
- both valid_start_date and valid_end_date for relationships could not be '12/31/2099', because in that case, these relationships are valid only for one day in 2099, which is not true.
- NHS Place of Admission and Discharge is invalid vocabulary_id, because it exceeds varchar(20) limit (please see checklist)
- valid_start_date for concepts in NHS Place of Admission and Discharge vocabulary is a default value of 1/1/1970, which is not a best practice.

**Comment by solmazeradat** [comment]
thanks @aostropolets for the points above. If there is anything I can do please let me know.

**Comment by TinyRickC137** [comment]
Included in the [latest release](https://athena.ohdsi.org/search-terms/terms?vocabulary=NHS+Ethnic+Category&vocabulary=NHS+Place+of+Service&page=1&pageSize=15&query=) with minor corrections (see my comment from above)

Release notes: https://github.com/OHDSI/Vocabulary-v5.0/releases/tag/v20230831_1693510428.000000
